analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp9.cs

```diff
@@ -294,4 +294,17 @@ public class Repro_8008
         (((_, _), _), null) => 44,
         (((_, _), _), _) => 45,
     };
+
+    public int TupleWithKnownNullConstraint()
+    {
+        string name1 = null;
+        string name2 = null;
+        return (name1, name2) switch
+        {
+            (null, null) => 0,  // FN
+            (null, _) => 1,     // FN
+            (_, null) => -1,    // FN
```

Comment:
uh... I expected your review update to turn these into TPs. Weird!

Reply:
That will need a different PR. It might be that the pattern matching isn't working correctly in this part of the switch expression:
```cs
var x = (a, b) switch // <- does the engine learn the necessary constraints here when dealing with a tuple?
{
   ...
};
```

Reply:
That's probably it. Good thinking!